### PR TITLE
Improve strings formatting PurchaserInfoStrings and PurchaseStrings

### PR DESCRIPTION
**Pull request by Juanpe**
Follow-up on #816 updates the format `PurchaserInfoStrings` and `PurchaseStrings`

### CODE REVIEW

File: Purchases/Logging/Strings/PurchaseStrings.swift [hunk 1]
```diff
@@ -13,40 +13,125 @@
...
+                "transaction, are you sure you want to do this?"
+
+        case .paymentqueue_removedtransaction(let transaction):
+            let errorUserInfo = (transaction.error as NSError?)?.userInfo ?? [:]
+            return "PaymentQueue removedTransaction: \(transaction.payment.productIdentifier) " +
+                "\(transaction.transactionIdentifier ?? "") " +
+                "(\(transaction.original?.transactionIdentifier ?? "") " +
+                "\(transaction.error?.localizedDescription ?? "") " +
+                "\(!errorUserInfo.isEmpty ? errorUserInfo.description : "") - " +
+                "\(transaction.transactionState.rawValue)"
```
Comment (aboedo) [inline Purchases/Logging/Strings/PurchaseStrings.swift:75]: this is soooo much easier to understand now 😻

File: Purchases/Purchasing/StoreKitWrapper.swift [hunk 2]
```diff
@@ -102,26 +98,15 @@ extension StoreKitWrapper: SKPaymentQueueDelegate {
...
     func paymentQueue(_ queue: SKPaymentQueue, removedTransactions transactions: [SKPaymentTransaction]) {
         for transaction in transactions {
-            Logger.debug(String(format: Strings.purchase.paymentqueue_removedtransaction,
-                                transaction.payment.productIdentifier,
-                                transaction.transactionIdentifier ?? "",
-                                transaction.original?.transactionIdentifier ?? "",
-                                transaction.error?.localizedDescription ?? "",
-                                (transaction.error as NSError?)?.userInfo ?? "",
-                                transaction.transactionState.rawValue))
+            Logger.debug(Strings.purchase.paymentqueue_removedtransaction(transaction: transaction))
```
Comment (aboedo) [inline Purchases/Purchasing/StoreKitWrapper.swift:109]: ❤️

**Review by aboedo — APPROVED** [review]

**Comment by aboedo** [comment]
fantastic as usual! merging now, thanks @Juanpe !!

**Comment by Juanpe** [comment]
A pleasure as always! 🙂

**Pull request merged**